TUM Campus App/CafeteriaMenuManager.swift

```diff
@@ -44,22 +48,91 @@ class CafeteriaMenuManager: Manager {
+            let newMenu = CafeteriaMenu(id: idNumber, date: dateAsDate, typeShort: typeShort, typeLong: typeLong, typeNr: nr, name: name, price: price, details: details)
             mensa.addMenu(newMenu)
             CafeteriaMenuManager.cafeteriaMenus.append(newMenu)
         }
     }
     
+    func parseMensaMenu(_ name: String) -> MenuDetail {
+        
+        let pattern = "(?:(?<=\\((?:[[a-z][A-Z][0-9]][[a-z][A-Z][0-9]]?,)?)|(?<=,(?:[[a-z][A-Z][0-9]][[a-z][A-Z][0-9]]?,)?))([[a-z][A-Z][0-9]][[a-z][A-Z][0-9]]?)(?=(?:,[[a-z][A-Z][0-9]][[a-z][A-Z][0-9]]?)*\\))"
+        var notMatchedEmoji: [String] = []
+        var matchedAnnotations: [String] = []
+        var matchedEmoji: [String] = []
+        var matchedDescriptions: [String] = []
+        let output = NSMutableString(string: name)
+        var startPoint = output.length
+        var withoutAnnotations = output
+        var withEmojiWithoutAnnotations = output
+
+        
+        if let regex = try? NSRegularExpression(pattern: pattern, options: []) {
+            
+            let matches = regex.matches(in: output as String, options: [], range: NSMakeRange(0, output.length))
+            
+            matchedAnnotations = matches.flatMap({output.substring(with: $0.range)})
+            
+            for match in matches {
+                
+                let matchedString = output.substring(with: match.range)
+                
+                startPoint = min(max(match.range.location - 1,0), startPoint)
+                
+                if let emoji = mensaAnnotationsEmoji[matchedString] {
+                    matchedEmoji.append(emoji)
+                    
+                } else {
+                    notMatchedEmoji.append("(\(matchedString))")
+                }
+                
+                if let matchedDescription = mensaAnnotationsDescription[matchedString] {
+                    matchedDescriptions.append(matchedDescription)
+                }
+            }
+            
+            regex.replaceMatches(in: output, options: [], range: NSMakeRange(0, output.length), withTemplate: "")
+            
+            let range = NSMakeRange(startPoint, output.length - startPoint)
+            
+            output.deleteCharacters(in: range)
+            withoutAnnotations = output
+            matchedEmoji.forEach({output.append($0)})
+            withEmojiWithoutAnnotations = output
+            notMatchedEmoji.forEach({output.append($0)})
+        }
+        return MenuDetail(name: String(output), nameWithoutAnnotations: String(withoutAnnotations), nameWithEmojiWithoutAnnotations: String(withEmojiWithoutAnnotations), annotations: matchedAnnotations, annotationDescriptions: matchedDescriptions)
+    }
+    
     func getURL() -> String {
         return "http://lu32kap.typo3.lrz.de/mensaapp/exportDB.php?mensa_id=all"
     }
     
 }
+
+struct Price {
+    
+    var st: Double
```

Comment:
Please never refer to the same naming scheme as in any api. We don't care how long our names are. We can rename things as much as we want ;)